cranelift/assembler-x64/meta/src/generate/format.rs

```diff
@@ -137,6 +140,22 @@ impl dsl::Format {
         }
     }
 
+    fn generate_vex(&self, f: &mut Formatter, vex: &dsl::Vex) {
+        f.empty_line();
+        f.comment("Emit VEX prefix.");
+        fmtln!(f, "let vex: VexInstruction<R> = VexInstruction::default();");
+        fmtln!(f, "let mut vex = vex.opcode(0x{:0x});", vex.opcodes.primary);
+        fmtln!(f, "vex.reg = self.xmm1.enc();");
+        fmtln!(f, "match &self.xmm_m128 {{");
+        fmtln!(f, "XmmMem::Xmm(r) => {{vex.rm =  XmmMem::Xmm(r.clone());}}");
+        fmtln!(f, "XmmMem::Mem(m) => {{vex.rm =  XmmMem::Mem(m.clone());}}");
+        fmtln!(f, "}}");
```

Comment:
When you're adding brace-delimited blocks, you can use `Formatter::add_block` to get the indentation right, avoid forgetting to add a closing brace, and to match the rest of this crate:

```suggestion
        f.add_block("match &self.xmm_m128", |f| {
            fmtln!(f, "XmmMem::Xmm(r) => {{vex.rm =  XmmMem::Xmm(r.clone());}}");
            fmtln!(f, "XmmMem::Mem(m) => {{vex.rm =  XmmMem::Mem(m.clone());}}");
        });
```

Reply:
This seems a bit redundant to match here in any case...